docs/fieldnorm/Software life cycle processes.md

```diff
@@ -0,0 +1,67 @@
+_-  Small-scale: the adaptation of a new component to existing software_\
+_-  Large-scale: developing entirely new software or big refactors of existing software_
+
+## 1.2.2 Requirements
+This process involves gathering, documenting, and analyzing the software requirements. It includes capturing both functional and non-functional requirements that the software must fulfill to meet the intended use and comply with regulatory standards. One example strategy for capturing this, which is also well suited for medical laboratories, is to use user stories because stakeholders within a medical laboratory have very specific needs but are often not able to translate those into well-defined technical requirements.
+
+Guidelines should support the following processes to gather, convert, track, and update the requirements during a project:
+
+  a.   Collaboration between end users and developers when defining software requirements.
+
+  b.  Convertion of users' requirements into small individual development tasks.
+
+  c.  The option to manage tasks using a task management system integrated with your Git development system.
+
+  d.  A plan for keeping in touch with end users throughout the different stages of the development process to avoid unexpected adjustments to requirements later on.
+
+_NOTE: A widely adopted software development strategy involves dividing a development project over several periods. It is advised to use an approach that enables the release of partial deliverables to users each period, which enables user feedback upon each deliverable during the entire lifetime of the project, and provides ample opportunities for clarification of requirements or adapt their priorities._
+
+A more thorough explanation of what requirements might include can be found in section (_link to "Requirements in software development"_) of this document.
+
+## 1.2.3 Design
+Here the software architecture is defined, describing the high-level structure, components, interfaces, and interactions of the software system. The architectural design lays the groundwork for the detailed design and implementation phases. 
+
+The following processes should be covered by the design phase:
+
+  a.  A coding policy that covers which language is used, how the language is formatted, do’s and don'ts, and the use of inline documentation. This policy is not to be strictly enforced but it will open discussions on how the team likes its code and lead to a more concise and unified coding strategy.
+
+  b.  Laboratories will have to document the architecture when the software is classified as class B or C according to IEC-62304.
+
+## 1.2.4 Development
+The development stage will be the bulk of a project; here, code is created that will eventually define the software. The following processes are not required, as smaller bioinformatics teams might find them too difficult to incorporate. Nonetheless, it is strongly recommended that your guidelines include:
```

Comment:
I would order than A, C and B, and then add the sentence about small teams before B.
I would say we all need a code management system and a well defined development and release cycle.  But also testing.

Maybe for B; we can add something like "In the case of a small bioinformatics team with only 1 or 2 members, it is recommended to think about what is feasible and realistic to do in terms of testing"

Reply:
I have reordered some of it and made c and d optional for smal to medium teams. if we feel like we really don't want the "small team" mentioned anywhere other than the introduction, we could remove it entirely.